screens/SendSetupScreen.js

```diff
@@ -144,29 +151,37 @@ const ErrorContainer = styled(View)`
   background-color: ${props => props.theme.danger700};
 `;
 
-const parseQr = (qrData: string): { address: string, amount: ?string } => {
-  let address = null;
-  let amount = null;
-
-  // Parse out address and any BIP21 relevant data
-  const parts = qrData.split("?");
-
-  address = parts[0];
-  const parameters = parts[1];
-  if (parameters) {
-    const parameterParts = parameters.split("&");
-    parameterParts.map(param => {
-      const [name, value] = param.split("=");
-      if (name === "amount") {
-        amount = value;
-      }
-    });
-  }
-  return {
-    address,
-    amount
-  };
-};
+// Only allow numbers and a single . in amount input
+const formatAmountInput = (amount: string, maxDecimals: number): string => {
+  const validCharacters = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0"];
+  let decimalCount = 0;
+
+  const valid = amount.split("").reduce((prev, curr, idx, array) => {
+    // Only allow max 1 leading 0
+    if (idx === 1 && curr === "0" && array[0] === "0") return prev;
+
+    // Filter non-valid characters
+    if (validCharacters.includes(curr)) return [...prev, curr];
+
+    // Max of 1 decimal
+    if (curr === "." && decimalCount === 0) {
+      decimalCount++;
+      return [...prev, curr];
+    }
+    return prev;
+  }, []);
+
+  // Add a 0 if first digit is a '.'
+  const maybeZero = valid[0] && valid[0] === "." ? ["0", ...valid] : valid;
+
+  // Restrict decimals
+  const decimalIndex = maybeZero.indexOf(".");
+  const decimalAdjusted =
+    decimalIndex >= 0
+      ? maybeZero.slice(0, decimalIndex + maxDecimals + 1)
+      : maybeZero;
+
+  return decimalAdjusted.join("");
```

Comment:
Missing closing paren, so doesn't compile.  
But should use this function from the utils, so once removed here that should be fixed also

Reply:
resolved from previous removal of function.

Reply:
Then why is it being re-added in this PR...?